drivers/clock_control/clock_control_nrf.c

```diff
@@ -714,24 +722,30 @@ static int cmd_status(const struct shell *shell, size_t argc, char **argv)
 	struct onoff_manager *lf_mgr =
 				get_onoff_manager(CLOCK_DEVICE,
 						  CLOCK_CONTROL_NRF_TYPE_LFCLK);
-	uint32_t abs_start, abs_stop;
+	uint32_t rel_start, rel_stop;
 	int key = irq_lock();
-	uint64_t now = k_uptime_get();
+	uint32_t now = z_nrf_rtc_timer_read();
 
 	(void)nrfx_clock_is_running(NRF_CLOCK_DOMAIN_HFCLK, (void *)&hfclk_src);
 	hf_status = (hfclk_src == NRF_CLOCK_HFCLK_HIGH_ACCURACY);
 
-	abs_start = hf_start_tstamp;
-	abs_stop = hf_stop_tstamp;
+	rel_start = (now - hf_start_tstamp) & Z_NRF_RTC_TIMER_MAX;
+	rel_stop = (now - hf_stop_tstamp) & Z_NRF_RTC_TIMER_MAX;
 	irq_unlock(key);
 
+	/* Convert to ms */
+	rel_start = (rel_start * 1000) / CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC;
+	rel_stop = (rel_stop * 1000) / CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC;
+
 	shell_print(shell, "HF clock:");
 	shell_print(shell, "\t- %srunning (users: %u)",
 			hf_status ? "" : "not ", hf_mgr->refs);
-	shell_print(shell, "\t- last start: %u ms (%u ms ago)",
-			(uint32_t)abs_start, (uint32_t)(now - abs_start));
-	shell_print(shell, "\t- last stop: %u ms (%u ms ago)",
-			(uint32_t)abs_stop, (uint32_t)(now - abs_stop));
+
+	if (IS_ENABLED(CONFIG_NRF_RTC_TIMER)) {
+		shell_print(shell, "\t- last start: %u ms ago", rel_start);
```

Comment:
I'm concerned that this will be completely wrong if the duration wraps the 24-bit RTC timer.